src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp

```diff
@@ -103,6 +628,36 @@ const RendererSettings& AppleseedRenderer::get_renderer_settings()
     return m_settings;
 }
 
+int AppleseedRenderer::NumRefs()
+{
+    return 1;
+}
+
+RefTargetHandle AppleseedRenderer::GetReference(int i)
+{
+    switch (i)
+    {
+      case 0:
+        return m_param_block;
+      default:
+        DbgAssert(false);
+        return nullptr;
+    }
+}
+
+
+void AppleseedRenderer::SetReference(int i, RefTargetHandle rtarg)
+{
+    switch (i)
+    {
+      case 0:
+        m_param_block = dynamic_cast<IParamBlock2*>(rtarg);
```

Comment:
This will have the effect of setting `m_param_block` to `nullptr` if `rtarg` is not a `IParamBlock2*`. Do you know if this can ever happen? If not, then I would suggest a `static_cast<>` instead.